query/compile.go

```diff
@@ -550,6 +556,161 @@ func (c *compiledField) compileMovingAverage(args []influxql.Expr) error {
 	}
 }
 
+func (c *compiledField) compileExponentialMovingAverage(name string, args []influxql.Expr) error {
+	if got := len(args) - 1; got < 1 || got > 3 {
```

Comment:
This should not use `len(args) - 1` as the error message will be very confusing.

This applies to every new function.